src/Product.php

```diff
@@ -615,6 +615,10 @@ protected function shouldProcessVariation(array $variation)
             return false;
         }
 
+        if (isset($variation['availableUntil']) && $this->isDateStillAvailable($variation['availableUntil'])) {
```

Comment:
I'm afraid this condition isn't quite correct. This would skip the variation if it's still available. I guess `!$this->isDateStillAvailable....`